Algorithm.CSharp/EmaCrossFuturesFrontMonthAlgorithm.cs

```diff
@@ -0,0 +1,128 @@
+/*
+ * QUANTCONNECT.COM - Democratizing Finance, Empowering Individuals.
+ * Lean Algorithmic Trading Engine v2.0. Copyright 2014 QuantConnect Corporation.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+*/
+
+using QuantConnect.Data;
+using QuantConnect.Data.Consolidators;
+using QuantConnect.Data.UniverseSelection;
+using QuantConnect.Indicators;
+using QuantConnect.Securities;
+using System.Linq;
+
+namespace QuantConnect.Algorithm.CSharp
+{
+    /// <summary>
+    /// This example demonstrates how to implement a cross moving average for the futures front contract
+    /// </summary>
+    /// <meta name="tag" content="using data" />
+    /// <meta name="tag" content="indicator" />
+    /// <meta name="tag" content="futures" />
+    public class EmaCrossFuturesFrontMonthAlgorithm : QCAlgorithm
+    {
+        private const decimal _tolerance = 0.001m;
+        private Symbol _symbol;
+        private ExponentialMovingAverage _fast;
+        private ExponentialMovingAverage _slow;
+        private IDataConsolidator _consolidator;
+
+        public override void Initialize()
+        {
+            SetStartDate(2013, 10, 08);
+            SetEndDate(2013, 10, 10);
+            SetCash(1000000);
+
+            var future = AddFuture(Futures.Metals.Gold);
+
+            // Only consider the front month contract
+            future.SetFilter(x => x.FrontMonth().OnlyApplyFilterAtMarketOpen());
```

Comment:
Might be worth adding a comment why is `OnlyApplyFilterAtMarketOpen` used? For performance I presume